render-only-sample/CubeTest/CubeTest_new.cpp

```diff
@@ -424,11 +424,154 @@ class D3DDefaultTexture
     std::shared_ptr<D3DDevice>              m_pDevice;
 };
 
+const char *DXGIFormatToString(DXGI_FORMAT format)
+{
+    switch (format)
+    {
+    case DXGI_FORMAT_R8G8B8A8_UNORM:
+        return "R8G8B8A8_UNORM";
+
+    case DXGI_FORMAT_R8G8_UNORM:
+        return "DXGI_FORMAT_R8G8_UNORM";
+
+    case DXGI_FORMAT_R8_UNORM:
+        return "DXGI_FORMAT_R8_UNORM";
+
+    case DXGI_FORMAT_A8_UNORM:
+        return "DXGI_FORMAT_A8_UNORM";
+
+    default:
+        return "UNKNOWN";
+    }
+}
+
 class D3DBitmapTexture
 {
 public:
 
-    D3DBitmapTexture(std::shared_ptr<D3DDevice> & inDevice)
+    const void *ConvertTextureFromRGBA(PBYTE pTexels, DXGI_FORMAT outFormat, ULONG texWidth, ULONG texHeight, D3D11_SUBRESOURCE_DATA &initData)
+    {
+
+        switch (outFormat)
+        {
+        case(DXGI_FORMAT_R8G8B8A8_UNORM) :
+        {
+            // Conversion is not needed.
+            initData.pSysMem = pTexels;
+            initData.SysMemPitch = texWidth * 4;
+            initData.SysMemSlicePitch = texWidth * 4 * texHeight;
+            return NULL;
+        }
+
+        case(DXGI_FORMAT_R8G8_UNORM) :
+        {
+            BYTE *colorConvertedBuffer = (BYTE*)malloc(texWidth * texHeight * sizeof(WORD) * 2);
+            if (colorConvertedBuffer == NULL)
```

Comment:
`operator new` will throw an exception if the allocation failed. It is not necessary to check the result of `operator new` for null.